portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/site/PortalSiteHelperImpl.java

```diff
@@ -330,20 +344,77 @@ public String getGatewaySiteId()
 	}
 
 	/**
+	 * {@inheritDoc}
+	 */
+	public String getSiteOrSectionTitle( Site site )
+	{
+		return getSiteOrSectionTitle( site, getProviderIDsForSite( site ) );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public String getSiteOrSectionTitle( Site site, List<String> providerIDs )
+	{
+		String retVal = cms.determineSiteOrSectionTitle( site, providerIDs );
```

Comment:
I think it can be the case the there isn't a CourseManagementService bean in some deployments, in these cases this will cause a throw a NullPointerException.